test/integration/query-interface.test.js

```diff
@@ -112,6 +113,114 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         expect(tableNames).to.include('my_test_table2');
       });
     }
+
+    if (dialect.supports.schemas) {
+      describe('schema support tests', () => {
+        const getSequelizeInstanceWithSchema = async () => {
+          const sequelize = createSequelizeInstance({ schema: 'schema_3' });
+          await sequelize.createSchema('schema_3');
+          await sequelize.queryInterface.createTable('schema_3_table_1', { name: DataTypes.STRING }, {
+            schema: 'schema_3',
+          });
+          await sequelize.queryInterface.createTable('schema_3_table_2', { name: DataTypes.STRING }, {
+            schema: 'schema_3',
+          });
+
+          await sequelize.createSchema('schema_4');
+          await sequelize.queryInterface.createTable('schema_4_table_1', { name: DataTypes.STRING }, {
+            schema: 'schema_4',
+          });
+          await sequelize.queryInterface.createTable('schema_4_table_2', { name: DataTypes.STRING }, {
+            schema: 'schema_4',
+          });
+
+          return { sequelize };
+        };
+
+        async function createTestTablesForSchema(schemaName) {
+          await this.queryInterface.createTable(`${schemaName}_table_1`, { name: DataTypes.STRING }, {
+            schema: schemaName,
+          });
+          await this.queryInterface.createTable(`${schemaName}_table_2`, { name: DataTypes.STRING }, {
+            schema: schemaName,
+          });
+        }
+
+        async function createSchemaAndTables(testSchemas = []) {
+          const baseTestSchemas = ['schema_1', 'schema_2', 'sequelize_test'].concat(testSchemas);
+
+          async function _createSchemaAndTables(schemaName) {
+            await this.sequelize.createSchema(schemaName);
+            await Reflect.apply(createTestTablesForSchema, this, [schemaName]);
+          }
+
+          await Promise.all(baseTestSchemas.map(_createSchemaAndTables.bind(this)));
+        }
+
+        const normalizeTableNames = (tableNames = []) => {
+          if (tableNames[0] && tableNames[0].tableName) {
+            return tableNames.map(v => v.tableName).sort();
+          }
+
+          return tableNames.sort();
+        };
+
+        it('shows all tables from the specified schema in the method options', async function () {
+          await createSchemaAndTables.apply(this);
+          const schemaOneTables = normalizeTableNames(await this.queryInterface.showAllTables({ schema: 'schema_1' }));
+          const schemaTwoTables = normalizeTableNames(await this.queryInterface.showAllTables({ schema: 'schema_2' }));
+
+          expect(schemaOneTables).to.include('schema_1_table_1');
```

Comment:
Use to.deep.equal instead of include, to make sure it contains that and only that